File: examples/python/conversation_example.py
"""Conversation domain example -- demonstrates create() and send() for multi-turn chat."""
import os
import sys
from onemin import OneMinClient, Models

api_key = os.environ.get("ONEMIN_API_KEY")
if not api_key:
    print("Set ONEMIN_API_KEY environment variable to run this example.")
    print("  export ONEMIN_API_KEY=your-key-here")
    sys.exit(1)

client = OneMinClient(api_key=api_key)

# --- Demo 1: Create a conversation ---
conv = client.conversation.create(
    title="SDK Demo Conversation",
    model=Models.Text.GPT_4O,
)
print("=== Conversation Created ===")
print(f"Conversation ID: {conv.conversation_id}")
print()

# --- Demo 2: Send messages in the conversation ---
reply = client.conversation.send(
    conv.conversation_id,
    "Hi! My name is Alice. Can you remember that?",
    model=Models.Text.GPT_4O,
)
print("=== First message ===")
print(f"Response: {reply.content}")
print()

reply = client.conversation.send(
    conv.conversation_id,
    "What is my name?",
    model=Models.Text.GPT_4O,
)
print("=== Follow-up message ===")
print(f"Response: {reply.content}")
print()

File: examples/typescript/conversation-example.ts
// Run: npx tsx conversation-example.ts
// conversation-example.ts -- demonstrates create() and send() for multi-turn chat
import { OneMinClient, Models } from 'onemin';

const apiKey = process.env.ONEMIN_API_KEY;
if (!apiKey) {
  console.error('Set ONEMIN_API_KEY environment variable to run this example.');
  console.error('  export ONEMIN_API_KEY=your-key-here');
  process.exit(1);
}

const client = new OneMinClient({ apiKey });

// --- Demo 1: Create a conversation ---
const conv = await client.conversation.create({
  title: 'SDK Demo Conversation',
  model: Models.Text.GPT_4O,
});
console.log('=== Conversation Created ===');
console.log(`Conversation ID: ${conv.conversationId}`);
console.log();

// --- Demo 2: Send messages in the conversation ---
const reply1 = await client.conversation.send(
  conv.conversationId,
  'Hi! My name is Alice. Can you remember that?',
  { model: Models.Text.GPT_4O },
);
console.log('=== First message ===');
console.log(`Response: ${reply1.content}`);
console.log();

const reply2 = await client.conversation.send(
  conv.conversationId,
  'What is my name?',
  { model: Models.Text.GPT_4O },
);
console.log('=== Follow-up message ===');
console.log(`Response: ${reply2.content}`);
console.log();
